rholang/src/main/scala/coop/rchain/rholang/interpreter/Runtime.scala

```diff
@@ -380,7 +381,7 @@ object Runtime {
       (reducer, replayReducer) = {
         lazy val replayDispatchTable: RhoDispatchMap[F] =
           dispatchTableCreator(
-            replaySpace,
+            ChargingRSpace.pureRSpace(replaySpace),
```

Comment:
Rather than creating the pureRspace ourselves, we should inject it from the outside. There should only be one call to `ChargingRSpace.pureRSpace` in prod code, and its output should find its way here.